src/presentation/http/http-server.ts

```diff
@@ -47,11 +48,16 @@ export default class HttpServer implements API {
    * @param domainServices - instances of domain services
    */
   public async run(domainServices: DomainServices): Promise<void> {
+    const middlewares = initMiddlewares();
```

Comment:
What if we will implement a middleware specific for some route? For example for checking a membership or something like that. For now, all router modules will know about all middlewares.

I think, only a particular route should know about middlewares it uses.

So maybe we don't need to instantiate it here, and just import them and use by each route? As i can see, authMiddleware is not a singletone, just a function. Correct?


```ts
// router/note.ts

import AuthMiddleware from '../middlewares/auth.ts'


fastify.post('/', { preHandler: [ AuthMiddleware ] }, async (request, reply) => {
 // now user is authorized
})
```

Reply:
I want to pass some service to authMiddleware, which will verify authorization tokens and get payload from it, that is why i can not import it.

Reply:
ok